test/jdk/java/util/stream/test/org/openjdk/tests/java/util/stream/CountTest.java

```diff
@@ -160,28 +180,52 @@ public void testOps(String name, TestData.OfDouble data) {
     public void testNoEvaluationForSizedStream() {
         {
             AtomicInteger ai = new AtomicInteger();
-            Stream.of(1, 2, 3, 4).peek(e -> ai.getAndIncrement()).count();
+            assertEquals(Stream.of(1, 2, 3, 4).peek(e -> ai.getAndIncrement()).count(), 4);
+            assertEquals(ai.get(), 0);
+
+            assertEquals(Stream.of(1, 2, 3, 4).peek(e -> ai.getAndIncrement()).parallel().count(), 4);
+            assertEquals(ai.get(), 0);
+
+            assertEquals(Stream.of(1, 2, 3, 4).peek(e -> ai.getAndIncrement())
+                .skip(1).limit(2).skip(1).count(), 1);
             assertEquals(ai.get(), 0);
 
-            Stream.of(1, 2, 3, 4).peek(e -> ai.getAndIncrement()).parallel().count();
+            assertEquals(Stream.of(1, 2, 3, 4).peek(e -> ai.getAndIncrement())
+                .parallel().skip(1).limit(2).skip(1).count(), 1);
             assertEquals(ai.get(), 0);
```

Comment:
Does it make sense to extract method here and launch it with different input streams?

Reply:
Well, it's not that easy as there are tests for IntStream, LongStream, DoubleStream and object Stream, so they cannot be generalized under the same method. Still extracting four methods reduces some amount of repetition, so I'll do this.